dialect/dialect.go

```diff
@@ -76,26 +76,34 @@ func Debug(d Driver, logger ...func(...interface{})) Driver {
 	if len(logger) == 1 {
 		logf = logger[0]
 	}
-	drv := &DebugDriver{d, func(_ context.Context, v ...interface{}) { logf(v...) }}
+	drv := &DebugDriver{d, func(_ context.Context, v LogEntry) { logf(v) }}
 	return drv
 }
 
 // DebugWithContext gets a driver and a logging function, and returns
 // a new debugged-driver that prints all outgoing operations with context.
-func DebugWithContext(d Driver, logger func(context.Context, ...interface{})) Driver {
+func DebugWithContext(d Driver, logger func(context.Context, LogEntry)) Driver {
```

Comment:
This introduces a breaking change. I'll try to think of an alternative how to add it without breaking users' code and still keep the API lean.